src/uwtools/rocoto.py

```diff
@@ -143,3 +67,276 @@ def validate_rocoto_xml(input_xml: OptionalPath) -> bool:
+            elif tag == STR.task:
+                self._add_task(e, val, taskname)
+            elif tag == STR.var:
+                for name, value in val.items():
+                    SubElement(e, STR.var, name=name).text = value
+
+    def _add_task(self, e: Element, config: dict, taskname: str) -> None:
+        """
+        Add a <task> element to the <workflow>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        :param taskname: The name of the task being defined.
+        """
+        e = SubElement(e, STR.task, name=taskname)
+        self._set_attrs(e, config)
+        self._set_and_render_jobname(config, taskname)
+        for tag in (
+            STR.account,
+            STR.command,
+            STR.cores,
+            STR.deadline,
+            STR.exclusive,
+            STR.jobname,
+            STR.join,
+            STR.memory,
+            STR.native,
+            STR.nodes,
+            STR.nodesize,
+            STR.partition,
+            STR.queue,
+            STR.rewind,
+            STR.shared,
+            STR.stderr,
+            STR.stdout,
+            STR.walltime,
+        ):
+            if tag in config:
+                SubElement(e, tag).text = config[tag]
+        for name, value in config.get(STR.envars, {}).items():
+            self._add_task_envar(e, name, value)
+        if STR.dependency in config:
+            self._add_task_dependency(e, config[STR.dependency])
+
+    def _add_task_dependency(self, e: Element, config: dict) -> None:
+        """
+        Add a <dependency> element to the <task>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        """
+        e = SubElement(e, STR.dependency)
+        for key, block in config.items():
+            tag, _ = self._tag_name(key)
+            if tag == STR.taskdep:
+                self._set_attrs(SubElement(e, STR.taskdep), block)
+            else:
+                raise UWConfigError("Unhandled dependency type %s" % tag)
+
+    def _add_task_envar(self, e: Element, name: str, value: str) -> None:
+        """
+        Add a <envar> element to the <task>.
+
+        :param e: The parent element to add the new element to.
+        :param config: Configuration data for this element.
+        """
+        e = SubElement(e, STR.envar)
+        SubElement(e, STR.name).text = name
+        SubElement(e, STR.value).text = value
+
+    def _add_workflow(self, config: dict) -> None:
+        """
+        Create the root <workflow> element.
+
+        :param config: Configuration data for this element.
+        """
+        config, e = config[STR.workflow], Element(STR.workflow)
+        self._set_attrs(e, config)
+        self._add_workflow_cycledefs(e, config[STR.cycledefs])
+        self._add_workflow_log(e, config[STR.log])
+        self._add_workflow_tasks(e, config[STR.tasks])
```

Comment:
Here's a case where the sub-config being passed down to an element-handling method is explicitly identified -- `config["dependency"]` in this case (but without using magic strings, which are factored out to the `STR` object).